testing/integration/tests/fil_integer_overflow.rs

```diff
@@ -60,94 +60,150 @@ fn integer_overflow() {
     // Instantiate machine
     tester.instantiate_machine(DummyExterns).unwrap();
```

Comment:
This test is mostly the same except that I split the overflow case into two:

1. Case 1: rely on overflow checks and expect the contract to panic/revert.
2. Case 2: explicitly specify wrapping semantics and make sure it the math works out.